engine/runtime/src/main/scala/org/enso/compiler/core/IR.scala

```diff
@@ -4723,6 +4722,7 @@ object IR {
       def mapExpressions(fn: Expression => Expression): Specified = {
         copy(
           name         = name.mapExpressions(fn),
+          ascribedType = ascribedType.map(fn),
```

Comment:
I guess this change of `IR.DefinitionArgument.Specified.mapExpressions` is answering [my question](https://github.com/enso-org/enso/pull/6584#discussion_r1188130798):

> I was hoping the `d.mapExpressions(resolveExpression ...)` will iterate over `IR.DefinitionArgument`, but it doesn't.

Overall the changes in this PR look more Scala-like than my #6584